server/index.js

```diff
@@ -150,6 +150,41 @@ app.post('/api/cart/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/orders', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!cartId) {
+    res.status(400).json({ error: 'no cart id found.' });
+  }
+  const { name, creditCard, shippingAddress } = req.body;
+  const errors = {};
+  if (!name) {
+    errors.name = 'missing or invalid name.';
+  }
+  if (!creditCard) {
+    errors.creditCard = 'missing or invalid credit card number.';
+  }
+  if (!shippingAddress) {
+    errors.shippingAddress = 'missing or invalid shipping address.';
+  }
+  if (Object.keys(errors).length) {
+    res.status(400).json(errors);
+  }
```

Comment:
Be sure to `return` here to prevent the rest of the code from running. If the server tries to respond twice, it'll crash or at least spew out a bunch of warnings and pollute our console.